### PR TITLE
add functions for saving and loading parameters for untyped models

### PR DESCRIPTION
**Pull request by SurajK7**
I found these functions useful for saving and loading weights. Are they any good or just redundant?

### CODE REVIEW

**Comment by junjihashimoto** [comment]
LGTM!

**Comment by SurajK7** [comment]
@junjihashimoto building upon this idea, can we convert weights from a pre-trained PyTorch model to a list of tensors as mentioned earlier and load them in a skeleton of the same model in hasktorch, this would facilitate in transfer learning as training models from scratch on imagenet data-set is a bit unfeasible for vision models. Does that sound good?

**Comment by austinvhuang** [comment]
@SurajK7 This kind of simple serialization sort of works ok especially staying within Haskell (analogous to pickling in python) but serialization w/ deeper interop, we're looking into torchscript.

See:
- https://github.com/hasktorch/hasktorch/pull/268
- https://github.com/hasktorch/hasktorch/pull/318

**Comment by SurajK7** [comment]
@austinvhuang I took a look at torchscript example earlier and from what I understood we can load pre-trained models and run inference in hasktorch but fine-tuning and transfer learning doesn't seem feasible via torchscript and I think that will greatly reduce the models' usability.

File: hasktorch/src/Torch/Serialize.hs [hunk 1]
```diff
@@ -3,9 +3,22 @@ module Torch.Serialize where
...
 
 load :: FilePath -> IO [Tensor]
 load file = cast1 S.load file
+
+saveWts :: Parameterized f => f -> FilePath -> IO ()
+saveWts model filePath = do 
+    let wts =  map (toDependent) (flattenParameters model)
+    save wts filePath
+
+loadSavedWts :: Parameterized b => b -> FilePath -> IO b
```
Comment (apaszke) [inline hasktorch/src/Torch/Serialize.hs]: _Weights_ is a very specific term used for only some parameters. Can we please call those functions `saveWeights` and `loadWeights`?

Comment (SurajK7) [reply hasktorch/src/Torch/Serialize.hs]: Thank you for pointing out, I am at fault, they should be better-called parameters I will fix that.

Comment (SurajK7) [reply hasktorch/src/Torch/Serialize.hs]: @apaszke I believe you meant to say something along the line of `saveParams` and `loadParams` right?
Also, can you shed some light on the earlier thread?

> @austinvhuang I took a look at torchscript example earlier and from what I understood we can load pre-trained models and run inference in hasktorch but fine-tuning and transfer learning doesn't seem feasible via torchscript and I think that will greatly reduce the models' usability.

Comment (apaszke) [reply hasktorch/src/Torch/Serialize.hs]: Oh yes, good point! I meant `saveParameters` and `loadParameters`. Sorry for the mixup

**Comment by junjihashimoto** [comment]
Thx!

**Pull request merged**